### PR TITLE
kie-issues#1285: Remove unnecessary base image from quarkus examples (quay.io/kiegroup/kogito-runtime-jvm:latest)

### PR DESCRIPTION
**Pull request by fantonangeli**
**Closes:** https://github.com/apache/incubator-kie-issues/issues/1285

This PR is related to:
- https://github.com/apache/incubator-kie-kogito-examples/pull/1978

### CODE REVIEW

File: packages/sonataflow-operator/bddframework/pkg/framework/util.go [hunk 1]
```diff
@@ -292,7 +292,11 @@ func AppendImageDefaultValues(image *api.Image) {
 	}
 
 	if len(image.Tag) == 0 {
-		image.Tag = GetKogitoImageVersion(version.GetTagVersion())
+		if image.Name == defaultRuntimeJVM {
+			image.Tag = GetKogitoImageVersion(version.GetOpenJDKImageVersion())
+		} else {
+			image.Tag = GetKogitoImageVersion(version.GetTagVersion())
+		}
```
Comment (tiagobento) [inline packages/sonataflow-operator/bddframework/pkg/framework/util.go]: `defaultRuntimeJVM` is a constant, so the `else` branch of this `if` statement is basically dead code...

File: packages/sonataflow-operator/version/version.go [hunk 2]
```diff
@@ -33,8 +33,15 @@ const (
 	tagVersion = "main"
 	// Kogito images tag version. Used for data-index and jobs-service images.
 	kogitoImagesTagVersion = "999-20240623"
+	// OpenJDK image version
+	openJDKImageVersion = "1.20"
```
Comment (tiagobento) [inline packages/sonataflow-operator/version/version.go]:
```suggestion
	openJDKImageTagVersion = "1.20"
```

I learned from @ricardozanini that `tag` is the entire thing, like `docker.io/apache/something:1.1.1`. In this case, tagVersion would be `1.1.1`.

File: packages/sonataflow-operator/version/version.go [hunk 3]
```diff
@@ -33,8 +33,15 @@ const (
...
 )
 
+// GetOpenJDKImageVersion gets the current OpenJDK image version.
+func GetOpenJDKImageVersion() string {
```
Comment (tiagobento) [inline packages/sonataflow-operator/version/version.go]:
```suggestion
func GetOpenJDKImageTagVersion() string {
```

**Review by tiagobento — CHANGES_REQUESTED** [review]
Thanks for addressing this @fantonangeli !

File: packages/sonataflow-operator/bddframework/pkg/framework/util.go (hunk 1 above)
Comment (fantonangeli) [reply packages/sonataflow-operator/bddframework/pkg/framework/util.go]: @tiagobento are you sure the `else` is dead code? 
I think `AppendImageDefaultValues()` is called with a different `image.Name` here:
https://github.com/apache/incubator-kie-tools/blob/main/packages/sonataflow-operator/bddframework/pkg/framework/kogitoserviceutils.go#L131

File: packages/sonataflow-operator/version/version.go (hunk 2 above)
Comment (fantonangeli) [reply packages/sonataflow-operator/version/version.go]: @tiagobento I used `1.20` as a value following other places in kie-tools and in the page below I can see it is a real tag:
https://catalog.redhat.com/software/containers/ubi9/openjdk-17/61ee7c26ed74b2ffb22b07f6/history
Anyway I can change it to `1.20-2` or `1.20-2.1719294794` of course.

File: packages/sonataflow-operator/bddframework/pkg/framework/util.go (hunk 1 above)
Comment (tiagobento) [reply packages/sonataflow-operator/bddframework/pkg/framework/util.go]: You're right, I apologize. Didn't realize this was part of a more generic method.

**Review by tiagobento — APPROVED** [review]

**Review by domhanak — APPROVED** [review]

**Review by domhanak — CHANGES_REQUESTED** [review]
In the referenced PR from examples I see;
`registry.access.redhat.com/ubi9/openjdk-17:1.18`

but in this PR I see:
`registry.access.redhat.com/ubi9/openjdk-17:1.20`

Please check, imho they should match

**Comment by fantonangeli** [comment]
@domhanak 
I used the "1.20" to keep consistency with other places in `kie-tools`, which uses "1.20", but I can change this PR to "1.18".

Otherwise, we need to update the version in `kogito-examples` where it is used in 18 places.
WDYT?

**Comment by tiagobento** [comment]
If we can upgrade to a newer version, then I think we should... @fantonangeli @domhanak... +1 for consistency too.

**Comment by domhanak** [comment]
@fantonangeli imho we update also examples if no issues, we should aim having this consistent

**Comment by fantonangeli** [comment]
@tiagobento @domhanak good, I will prepare the kogito-example update PR

**Review by tiagobento — APPROVED** [review]

**Comment by fantonangeli** [comment]
`kogito-examples` update PR: https://github.com/apache/incubator-kie-kogito-examples/pull/1980

**Review by domhanak — APPROVED** [review]

**Pull request merged**